[WIP] feat: implement comment-check in workflows.

mo3et opened the pull request:
<!--
🎉 Thanks for sending a pull request to Kratos! Here are some tips for you:

1. If this is your first time contributing to Kratos, please read our contribution guide: https://go-kratos.dev/en/docs/community/contribution/
2. Ensure you have added or ran the appropriate tests and lint for your PR, please use `make lint` and `make test` before filing your PR, use `make clean` to tidy your go mod.
3. If the PR is unfinished, you may need to mark it as a WIP(Work In Progress) PR or Draft PR
4. Please use a semantic commits format title, such as `<type>[optional scope]: <description>`, see: https://go-kratos.dev/docs/community/contribution#type
5. at the same time, please note that similar work should be submitted in one PR as far as possible to reduce the workload of reviewers. Do not split a work into multiple PR unless it should.
-->

<!--
🎉 感谢您向 Kratos 发送 PR！以下是一些提示：
如果这是你第一次为 Kratos 贡献，请阅读我们的贡献指南：https://go-kratos.dev/en/docs/community/contribution/
2、确保您已经为您的 PR 添加或运行了适当的测试和lint，请在提交PR之前使用“make lint”和“make test”，使用“make clean”整理您的 go.mod。
3、如果 PR 未完成，您可能需要将其标记为 WIP（Work In Progress）PR 或 Draft PR
4、请使用语义提交格式标题，如“<类型>[可选范围]：<说明>`，请参阅：https://go-kratos.dev/docs/community/contribution#type
5. 同时请注意，同类的工作请尽量在一个PR中提交，以减轻 review 者的工作负担，不要把一项工作拆分成很多个PR，除非它应该这样做。
-->


#### Description (what this PR does / why we need it):
<!--
* The description should include the motivation for this PR or contrast this with previous behavior
-->
implement comment check in workflows. When commit PR have Chinese. CI have a failed.

#### Which issue(s) this PR fixes (resolves / be part of):
<!--
* Automatically closes linked issue when PR is merged.
* If your PR is not fully resolved the issue, please use `part of #<issue number>` instead.

Usage: `fixes/resolves #<issue number>`, or `fixes/resolves (paste link of issue)`.
-->

#3422 

#### Other special notes for the reviewers:
<!--
* Somethings that need extra attention for the reviewers
* Some additional notes, TODO list, etc.
-->

Need remove test Chinese comments when PR is merged.

shenqidebaozi commented:
It seems to be effective. Does he check the entire code every time? Also, can we directly comment on the issue in the PR, otherwise we still need to check the action details

mo3et commented:
> It seems to be effective. Does he check the entire code every time? Also, can we directly comment on the issue in the PR, otherwise we still need to check the action details

This Check only run when pull request into `main` branch or `workflow_dispatch`. 

I think it is feasible to have the bot comment on the error details. I will add this part in a later commit.

mo3et commented:
> What I mean is, can we only check the code for PR changes? Instead of scanning the full code

I tried to implement checking only the diff, and it works. However, I think this approach ignores existing errors and doesn’t handle them properly. Since this CI process only scans text and isn’t time-consuming, I recommend scanning the entire repo.

If you wish, I will implement a version that checks only the diff.

mo3et commented:
I’ve updated the version to only check the diff files in the PR and commented out the whole repo check version.

I also added a bot comment when the check fails. If you think it looks good, I’ll push the final version.

mo3et commented:
I try to use `secrets.GITHUB_TOKEN` and `secrets.BOT_GITHUB_TOKEN`. It is not work. I think it this PAT is lose **issues:write** and **pull-requests: write** permissions.  I try to in our repos is success. 

I need to help provide the Bot [PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) having **issues:write** and **pull-requests: write**. Like `secrets.BOT_GITHUB_TOKEN` but have more premissions.


---
 - **Test comment effect when CI failed in our repo**

<img width="1050" alt="image" src="https://github.com/user-attachments/assets/5aca9d21-09b4-413f-a5b2-164dd3a5606f">

- **bot comment CI failed details**

<img width="1438" alt="image" src="https://github.com/user-attachments/assets/c9fdea36-a2ed-494c-8bf1-ed6fefe55898">

shenqidebaozi commented:
Understood, let me add a key

mo3et commented:
> Understood, let me add a key

Sorry, my bad. It just use **ci_bot_token** using translator issue. 

But use `pull_re`

mo3et commented:
同时也辛苦包子哥哥这几天的跟进和回复 感谢！

mo3et commented:
@shenqidebaozi @Windfarer @daemon365

Hello, can your have a check this PR?
